js/components/search.js

```diff
@@ -5,85 +5,165 @@ $( document ).ready(function() {
-            content_type = '<span class="glyphicon glyphicon-file"></span> ' + hit._source.content_type + '';
-          }
-          $('.search-results-container').append('<article class="search-result-list"><h1><a href="'
-            + hit._source.url
-            + '" target="_blank">'
-            + hit._source.title
-            + '</a></h1>'
-            + '<p>'
-            + result_description
-            + '</p>'
-            +'<p>Tagged&nbsp;/'
-            + tags
-            + '</p>'
-            + '</article>');
-        });
-      });
+  /**
+   * Returns URL for search request
+   *
+   * @param {string} query - Raw string holding search terms/query
+   * @param {string} apiKey - Raw string holding API key for api.data.gov
+   * @param {object} [options] - Optional search parameters
+   * @param {number} [options.size=10] - How many results to fetch
+   * @param {number} [options.from=0] - Offset to start from for paging
+   * @returns {string} URL for desired search
+   */
+  function apiSearchUrl( query, apiKey, options ) {
+    var q = encodeURIComponent( query );
+    var size = (options && options.size) || 10;
+    var from = (options && options.from) || 0;
+
+    return (
+      'https://api.data.gov/beckley-federalist/v0/resources/eiti/?' +
+      'q=' + q + '&size=' + size + '&from=' + from + '&api_key=' + apiKey
+    );
+  }
+
+  function localSearchUrl( query, options ) {
+    var q = encodeURIComponent( query );
+    var size = (options && options.size) || 10;
+    var from = (options && options.from) || 0;
+
+    return '/search-results/?q=' + q + '&size=' + size + '&from=' + from;
   }
-  else {
+
+  /**
+   * Returns HTML string of Glyph for given content type
+   *
+   * @param {string} contentType - given document content type
+   * @returns {string} glyph and label for content type or generic default
+   */
+  function getContentTypeGlyph(contentType) {
+    var contentTypes = {
+      'text/html':       '<span class="glyphicon glyphicon-link"></span> Website',
+      'application/pdf': '<span class="glyphicon glyphicon-file"></span> PDF',
+      'default':         '<span class="glyphicon glyphicon-file"></span> ' + contentType
+    };
+
+    return contentTypes[contentType]
+      ? contentTypes[contentType]
+      : contentTypes['default']
+  }
+
+  var query = getQueryParam('q'),
+      size  = parseInt(getQueryParam('size'), 10) || 10,
+      from  = parseInt(getQueryParam('from'), 10) || 0;
+
+  $('.search-string').text( query );
+  $('.site-search-text').attr('value', query );
+
+  if ( ! query ) {
     $('.search-no-results').show();
     $('.loading').remove();
     $('.search-results-count').append( 0 + ' search results');
-
+    return;
   }
+
+  $('input.q').val( query );
+  $('.search-result-list').html(
+    '<div class="loading">' +
+      '<span class="glyphicon glyphicon-refresh"></span> Loading' +
+    '</div>'
+  );
+
+  $.ajax({
+    url: apiSearchUrl( query, eiti.beckleyApiKey, {
```

Comment:
we didn't appear to use the API key anywhere but here, and I don't like leaving around variables for the sake of having variables, so I inlined this here. I did this a few other places too